go-controller/pkg/ovn/healthcheck/egressip_healthcheck.go

```diff
@@ -0,0 +1,166 @@
+package healthcheck
+
+import (
+	"fmt"
+	"net"
+	"sync"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"k8s.io/klog/v2"
+)
+
+const (
+	ServiceEgressIpNode = "Service_Egress_IP"
+)
+
+// UnimplementedHealthServer must be embedded to have forward compatible implementations.
+type healthServer struct {
+	UnimplementedHealthServer
+}
+
+func (healthServer) Check(_ context.Context, req *HealthCheckRequest) (*HealthCheckResponse, error) {
+	response := HealthCheckResponse{}
+
+	if req.GetService() == ServiceEgressIpNode {
+		response.Status = HealthCheckResponse_SERVING
+	} else {
+		response.Status = HealthCheckResponse_NOT_SERVING
+	}
+	return &response, nil
+}
+
+type egressIPHealthServer struct {
+	// Management port bound by server
+	node_mgmt_ip net.IP
+
+	// EgressIP Node reachability gRPC port (0 means it should use dial instead)
+	health_check_port int
+}
+
+func NewEgressIPHealthServer(node_mgmt_ip net.IP, health_check_port int) (*egressIPHealthServer, error) {
+	return &egressIPHealthServer{
+		node_mgmt_ip:      node_mgmt_ip,
+		health_check_port: health_check_port,
+	}, nil
+}
+
+func (ehs *egressIPHealthServer) Run(stopCh <-chan struct{}) {
+	lis, err := net.Listen("tcp", fmt.Sprintf("%s:%d", ehs.node_mgmt_ip.String(), ehs.health_check_port))
+	if err != nil {
+		klog.Fatalf("Health checking listen failed: %v", err)
+		return
+	}
+
+	wg := &sync.WaitGroup{}
+
+	// TODO (FF): add TLS credentials support
+	grpcServer := grpc.NewServer()
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		RegisterHealthServer(grpcServer, &healthServer{})
+		klog.Infof("Starting Egress IP Health Server on %s:%d", ehs.node_mgmt_ip.String(), ehs.health_check_port)
+		if err := grpcServer.Serve(lis); err != nil && err != grpc.ErrServerStopped {
+			klog.Fatalf("Health checking serve failed: %v", err)
+		}
+		klog.Infof("Stopped Egress IP Health Server on %s:%d", ehs.node_mgmt_ip.String(), ehs.health_check_port)
+	}()
+
+	<-stopCh
+
+	klog.Info("Shutting down Egress IP Health Server")
+	grpcServer.Stop()
+	wg.Wait()
+	klog.Info("Shut down Egress IP Health Server")
+}
+
+type EgressIPHealthClient interface {
+	IsConnected() bool
+	Connect(dial_ctx context.Context, mgmtIPs []net.IP, health_check_port int) bool
+	Disconnect()
+	Probe(dial_ctx context.Context) bool
+}
+
+type egressIPHealthClient struct {
+	nodeName string // debug
```

Comment:
what does `// debug` mean?

Reply:
heh, it is a silly way of saying that it is there to make logs easier to read; there is no real need to have this attribute in the struct. I always struggle a bit between making logs helpful at the cost of attributes like this.
All in all, we have a decision to make: 1) leave nodeName here to help on logs or 2) remove it completely. wdyt?

Reply:
I removed the `//debug` for now. Hope that makes it less confusing.